components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/connector/recaptcha/EmailOTPCaptchaConnector.java

```diff
@@ -0,0 +1,135 @@
+import org.wso2.carbon.identity.captcha.util.CaptchaUtil;
+import org.wso2.carbon.identity.governance.IdentityGovernanceException;
+import org.wso2.carbon.identity.governance.IdentityGovernanceService;
+
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+
+import static org.wso2.carbon.identity.captcha.util.CaptchaConstants.SSO_LOGIN_RECAPTCHA_CONNECTOR_NAME;
+
+/**
+ * Recaptcha Connector for Email OTP.
+ */
+public class EmailOTPCaptchaConnector extends AbstractReCaptchaConnector {
+
+    private static final Log log = LogFactory.getLog(EmailOTPCaptchaConnector.class);
+    private static final String SECURED_DESTINATIONS = "/commonauth";
+    public static final String EMAIL_OTP_AUTHENTICATOR_NAME = "email-otp-authenticator";
+    public static final String IS_REDIRECT_TO_EMAIL_OTP = "isRedirectToEmailOTP";
+    public static final String RESEND_CODE = "resendCode";
+
+    private IdentityGovernanceService identityGovernanceService;
+
+    @Override
+    public void init(IdentityGovernanceService identityGovernanceService) {
+
+        this.identityGovernanceService = identityGovernanceService;
+    }
+
+    @Override
+    public int getPriority() {
+        return 30;
+    }
+
+    @Override
+    public boolean canHandle(ServletRequest servletRequest, ServletResponse servletResponse) throws CaptchaException {
+
+        String path = ((HttpServletRequest) servletRequest).getRequestURI();
+
+        if (StringUtils.isBlank(path) || !(CaptchaUtil.isPathAvailable(path, SECURED_DESTINATIONS))) {
+            return false;
+        }
+
+        String sessionDataKey = servletRequest.getParameter(FrameworkUtils.SESSION_DATA_KEY);
+        AuthenticationContext context = FrameworkUtils.getAuthenticationContextFromCache(sessionDataKey);
+
+        if (context == null
+                || !EMAIL_OTP_AUTHENTICATOR_NAME.equals(context.getCurrentAuthenticator())
+                || (EMAIL_OTP_AUTHENTICATOR_NAME.equals(context.getCurrentAuthenticator())
+                && !Boolean.parseBoolean((String) context.getProperty(IS_REDIRECT_TO_EMAIL_OTP)))) {
+            return false;
+        }
+
+        if (Boolean.parseBoolean(servletRequest.getParameter(RESEND_CODE))) {
+            return false;
+        }
+
+        return isEmailRecaptchaEnabled(servletRequest);
+    }
+
+    @Override
+    public CaptchaPreValidationResponse preValidate(ServletRequest servletRequest, ServletResponse servletResponse) {
+
+        CaptchaPreValidationResponse preValidationResponse = new CaptchaPreValidationResponse();
+        preValidationResponse.setCaptchaValidationRequired(true);
+        return preValidationResponse;
+    }
+
+    @Override
+    public CaptchaPostValidationResponse postValidate(ServletRequest servletRequest, ServletResponse servletResponse) {
+
+        return null;
+    }
+
+    public boolean isEmailRecaptchaEnabled(ServletRequest request) {
```

Comment:
Isn't this exactly same as the one in SSO login captcha? https://github.com/wso2-extensions/identity-governance/blob/master/components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/connector/recaptcha/SSOLoginReCaptchaConfig.java#L328-L369

Shall we move it to a common place and reuse?

Reply:
Updated the method to check only the configs because the other checks are done in the canHandle method. It doesn't contain the same logic as the given SSO Login captcha method.